src/astarte.rs

```diff
@@ -0,0 +1,67 @@
+/*
+ * This file is part of Edgehog.
+ *
+ * Copyright 2022 SECO Mind Srl
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+use astarte_sdk::builder::AstarteOptions;
+use astarte_sdk::{AstarteError, AstarteSdk};
+use async_trait::async_trait;
+use mockall::automock;
+use serde::Serialize;
+
+#[automock]
+#[async_trait]
+pub trait Publisher: Send + Sync {
+    async fn send_object<T: 'static>(
+        &self,
+        interface_name: &str,
+        interface_path: &str,
+        data: T,
+    ) -> Result<(), AstarteError>
+    where
+        T: serde::Serialize + Send;
+}
```

Comment:
Also, the trait should probably support additional methods (e.g. `send_object_with_timestamp` and `send`). Are they going to be added in future PRs? In that case it would probably be nice to leave a TODO